### ISSUE TITLE
Hybrid Android - App crashes on launching

### ISSUE THREAD

**Issue by IuliiaHerets** [Expensify/App#53987]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.75-0
**Reproducible in staging?:** Yes
**Reproducible in production?:** No
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:** N/A
**If this was caught during regression testing, add the test name, ID and link from TestRail:** Exp
**Email or phone of affected tester (no customers):** N/A
**Issue reported by:** Applause Internal Team
**Device used:** Samsung Galaxy Z Fold 4 / Android 14
**App Component:** Other

## Action Performed:
1. Launch hybrid Android app.

## Expected Result:
App will not crash.

## Actual Result:
App crashes.

## Workaround:
Unknown

## Platforms:
- [ ] Android: Standalone
- [x] Android: HybridApp
- [ ] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos
[Bug6691708_1733977412090!log.txt](https://github.com/user-attachments/files/18107235/Bug6691708_1733977412090.log.txt)

https://github.com/user-attachments/assets/246cb5a3-9e83-4461-b887-c9abd6cc8040

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by IuliiaHerets** [comment]
Our prod version from PlayStore is 9.0.73-8 

https://github.com/user-attachments/assets/3d65c2ed-9314-41d4-b7b6-e04d53f81fe2

**Comment by lakchote** [comment]
Raised a discussion [here](https://expensify.slack.com/archives/C05LX9D6E07/p1733994042262189?thread_ts=1733918598.619389&cid=C05LX9D6E07).

**Comment by Julesssss** [comment]
Firebase crash [logs](https://console.firebase.google.com/u/0/project/expensify-mobile-app/crashlytics/app/android:org.me.mobiexpensifyg/issues/4762b53d618e22b55b6a312898169877?time=last-seven-days&types=crash&versions=9.0.75.0%20(509007500)&sessionEventKey=675A8EDC01DF000104387244A03EFB93_2025867617492663644). 

```
Fatal Exception: java.lang.RuntimeException: undefined is not an object (evaluating 'Store.ReportHistory.bindCacheClearingEvents') at position 81:24 in app/expensify.js
Expensify@app/expensify.js:81:24
getInstance@app/expensify.js:590:44
main@app/main.js:4:36
onAllFilesIncluded@app/yapl/Yapl.js:244:21
onAllFilesIncluded@[native code]
yaplIncludeFiles@app/yapl/Yapl.js:51:17
init@app/yapl/Yapl.js:246:25
global code@YAPL-Platform:1:10
       at yapl.android.Yapl.handleJSException(Yapl.java:624)
       at yapl.android.Yapl.js_evaluate(Yapl.java:573)
       at yapl.android.Yapl.js_evaluate(Yapl.java:556)
       at yapl.android.Yapl.setActiveActivity(Yapl.java:765)
       at yapl.android.YaplActivityBase.onStart(YaplActivityBase.java:379)
       at android.app.Instrumentation.callActivityOnStart(Instrumentation.java:1583)
       at android.app.Activity.performStart(Activity.java:8647)
       at android.app.ActivityThread.handleStartActivity(ActivityThread.java:3826)
       at android.app.servertransaction.TransactionExecutor.performLifecycleSequence(TransactionExecutor.java:230)
       at android.app.servertransaction.TransactionExecutor.cycleToPath(TransactionExecutor.java:210)
       at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:182)
       at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:103)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2462)
       at android.os.Handler.dispatchMessage(Handler.java:106)
       at android.os.Looper.loopOnce(Looper.java:205)
       at android.os.Looper.loop(Looper.java:294)
       at android.app.ActivityThread.main(ActivityThread.java:8256)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:557)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:993)
        
```

**Comment by Julesssss** [comment]
We've seen this Yapl issue [before](https://expensify.slack.com/archives/C01GTK53T8Q/p1732719415515299?thread_ts=1730223599.479179&cid=C01GTK53T8Q) and I'm almost certain it's because we're failing to build JS. Here's what we need to be doing as part of the build process.

```
11-27 15:49:02.427 10958 10958 E AndroidRuntime: java.lang.RuntimeException: undefined is not an object (evaluating 'Store.ReportHistory.bindCacheClearingEvents') at position 81:24 in app/expensify.js
```

**Fix:**
`npm run grunt:build:shared`

**Comment by Julesssss** [comment]
Here's my [WIP PR](https://github.com/Expensify/App/pull/54032/files). I think some small path changes here will resolve the issue 🤞

**Comment by Julesssss** [comment]
- [Fix PR merged](https://github.com/Expensify/App/pull/54028)
- Test [run](https://github.com/Expensify/App/actions/runs/12298516942) of new `build.yml` file

**Comment by Julesssss** [comment]
Fixed as of `9.0.75.1`.

![Screenshot_20241212_151040](https://github.com/user-attachments/assets/e7ac7207-ad61-4c7f-86ca-9412c6789fa3)

**Comment by Julesssss** [comment]
QA verified fixed [here](https://expensify.slack.com/archives/C9YU7BX5M/p1734018542662329?thread_ts=1734016418.576979&cid=C9YU7BX5M)
